### PR TITLE
dog: allow to specify hostnames in dog commands

### LINKED ISSUE #47: command option 'localhost' support
when I run `dog vdi list -a localhost`, it displays:
\# Invalid ip address localhost
It seems that sheepdog doesnot support localhost option.
And I think the option is necessary. Some up-level program may generate `dog vdi create XXXX -a localhost` command when using sheepodog as backend suppport.
So I suggest maybe we can add 'localhost' option support in dog.c. Just add a simple judge statement in "getopt_long switch case 'a' statment" works well. Maybe there is better solution.
Thanks.

### PR DESCRIPTION
**Pull request by vtolstov**
fixes #47

Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

File: lib/net.c [hunk 1]
```diff
@@ -416,14 +416,45 @@ char *sockaddr_in_to_str(struct sockaddr_in *sockaddr)
...
+	hints.ai_flags = 0;
+	hints.ai_protocol = 0;
 
-	if (af == AF_INET) {
-		addr_start_idx = 12;
-		memset(addr, 0, addr_start_idx);
+	res = getaddrinfo(ipstr, NULL, &hints, &result);
+	if (res != 0) {
+		return NULL;
 	}
```
Comment (mitake) [inline lib/net.c]: Please emit {} if the branch have only one statement.

File: lib/net.c [hunk 2]
```diff
@@ -416,14 +416,45 @@ char *sockaddr_in_to_str(struct sockaddr_in *sockaddr)
...
+			addr = memcpy(addr+addr_start_idx, &sin6->sin6_addr, rp->ai_addrlen);
+			break;
+		}
+		break;
+	}
+
+	if (rp == NULL) {
 		return NULL;
+	}
+
```
Comment (mitake) [inline lib/net.c]: Please emit {} if the branch have only one statement.

**Comment by mitake** [comment]
Thanks, I added some trivial comments. Could you update?

**Comment by vtolstov** [comment]
I'm fix patch. Please, merge.

**Comment by mitake** [comment]
Pulled, thanks.

**Pull request merged**